bigtable/tests/unit/test_instance.py

```diff
@@ -408,6 +408,304 @@ def test_list_tables_failure_name_bad_before(self):
+        client = self._make_client(project=self.PROJECT,
+                                   credentials=credentials, admin=True)
+        instance = self._make_one(self.INSTANCE_ID, client, self.LOCATION_ID)
+
+        name = 'name3373707'
+        etag = 'etag3123477'
+        description = 'description-1724546052'
+        expected_response = {
+            'name': name,
+            'etag': etag,
+            'description': description
+        }
+        expected_response = instance_data_v2_pb2.AppProfile(
+            **expected_response)
+
+        response_pb = instance_data_v2_pb2.AppProfile(
+            name=name,
+            etag=etag,
+            description=description
+        )
+
+        # Patch the stub used by the API method.
+        client._instance_admin_client = instance_api
+        bigtable_instance_stub = (
+            client._instance_admin_client.bigtable_instance_admin_stub)
+        bigtable_instance_stub.CreateAppProfile.side_effect = [response_pb]
+
+        # Perform the method and check the result.
+        app_profile_id = 'appProfileId1262094415'
+        result = instance.create_app_profile(app_profile_id=app_profile_id,
+                                             routing_policy_type=1)
+
+        self.assertEqual(result, expected_response)
+
+    def test_create_app_profile_with_single_routing_policy(self):
+        from google.cloud.bigtable_admin_v2.proto import (
+            instance_pb2 as instance_data_v2_pb2)
+        from google.cloud.bigtable_admin_v2.gapic import (
+            bigtable_instance_admin_client)
+
+        instance_api = (
+            bigtable_instance_admin_client.BigtableInstanceAdminClient(
+                mock.Mock()))
+
+        credentials = _make_credentials()
+        client = self._make_client(project=self.PROJECT,
+                                   credentials=credentials, admin=True)
+        instance = self._make_one(self.INSTANCE_ID, client, self.LOCATION_ID)
+
+        name = 'name3373707'
+        etag = 'etag3123477'
+        description = 'description-1724546052'
+        cluster_id = 'cluster-id'
+        expected_response = {
+            'name': name,
+            'etag': etag,
+            'description': description
+        }
+        expected_response = instance_data_v2_pb2.AppProfile(
+            **expected_response)
+
+        response_pb = instance_data_v2_pb2.AppProfile(
+            name=name,
+            etag=etag,
+            description=description
+        )
+
+        # Patch the stub used by the API method.
+        client._instance_admin_client = instance_api
+        bigtable_instance_stub = (
+            client._instance_admin_client.bigtable_instance_admin_stub)
+        bigtable_instance_stub.CreateAppProfile.side_effect = [response_pb]
+
+        # Perform the method and check the result.
+        app_profile_id = 'appProfileId1262094415'
+        result = instance.create_app_profile(app_profile_id=app_profile_id,
```

Comment:
Can you please add a test to make sure that the request proto you expect is sent to the stub?

Reply:
@sduskis Working on mock stub calls in unit tests